src/services/codefixes/fixAddMissingParam.ts

```diff
@@ -0,0 +1,248 @@
+    isAccessExpression,
+    isCallExpression,
+    isIdentifier,
+    isParameter,
+    isPropertyDeclaration,
+    isSourceFileFromLibrary,
+    isVariableDeclaration,
+    last,
+    lastOrUndefined,
+    length,
+    map,
+    MethodDeclaration,
+    Node,
+    NodeBuilderFlags,
+    ParameterDeclaration,
+    Program,
+    SignatureKind,
+    sort,
+    SourceFile,
+    SyntaxKind,
+    textChanges,
+    Type,
+    TypeChecker,
+} from "../_namespaces/ts";
+import {
+    codeFixAll,
+    createCodeFixAction,
+    registerCodeFix,
+} from "../_namespaces/ts.codefix";
+
+const fixId = "addMissingParam";
+const errorCodes = [Diagnostics.Expected_0_arguments_but_got_1.code];
+
+registerCodeFix({
+    errorCodes,
+    fixIds: [fixId],
+    getCodeActions(context) {
+        const info = getInfo(context.sourceFile, context.program, context.span.start);
+        if (info === undefined) return undefined;
+
+        const changes = textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, info));
+        return [createCodeFixAction(fixId, changes, info.description, fixId, Diagnostics.Add_all_missing_parameters)];
+    },
+    getAllCodeActions: context =>
+        codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(context.sourceFile, context.program, diag.start);
+            if (info) {
+                doChange(changes, context.sourceFile, info);
+            }
+        }),
+});
+
+type ConvertableSignatureDeclaration =
+    | FunctionDeclaration
+    | FunctionExpression
+    | ArrowFunction
+    | MethodDeclaration;
+
+interface SignatureInfo {
+    readonly description: DiagnosticOrDiagnosticAndArguments;
+    readonly newParameters: ParameterInfo[];
+    readonly declarations: ConvertableSignatureDeclaration[];
+    readonly overload: ConvertableSignatureDeclaration | undefined;
+}
+
+interface ParameterInfo {
+    readonly pos: number;
+    readonly declaration: ParameterDeclaration;
+}
+
+function getInfo(sourceFile: SourceFile, program: Program, pos: number): SignatureInfo | undefined {
+    const token = getTokenAtPosition(sourceFile, pos);
+    const callExpression = findAncestor(token, isCallExpression);
+    if (callExpression === undefined || length(callExpression.arguments) === 0) return undefined;
+
+    const checker = program.getTypeChecker();
+    const type = checker.getTypeAtLocation(callExpression.expression);
+    const signatures = checker.getSignaturesOfType(type, SignatureKind.Call);
+    const signature = lastOrUndefined(sort(signatures, (a, b) => length(a.parameters) - length(b.parameters)));
+    if (
+        signature &&
+        signature.declaration &&
+        isConvertableSignatureDeclaration(signature.declaration)
+    ) {
+        const declaration = (signature.declaration.body === undefined ? find(signature.declaration.symbol.declarations, d => isConvertableSignatureDeclaration(d) && !!d.body) : signature.declaration) as ConvertableSignatureDeclaration;

```

Comment:
I'd suggest a comment in front like  `// find a non-overload signature`, or just rename `declaration` to `nonOverloadDeclaration`